src/GraphQL.Tests/Types/AutoRegisteringInterfaceGraphTypeTests.cs

```diff
@@ -294,7 +295,8 @@ public async Task Argument_ResolverTests_WithNonNullString(string fieldName, str
         using var provider = serviceCollection.BuildServiceProvider();
         context.RequestServices = provider;
         fieldType.Resolver.ShouldNotBeNull();
-        (await fieldType.Resolver!.ResolveAsync(context).ConfigureAwait(false)).ShouldBe(expected);
+        var ex = await Should.ThrowAsync<InvalidOperationException>(async () => await fieldType.Resolver!.ResolveAsync(context).ConfigureAwait(false)).ConfigureAwait(false);
+        ex.Message.ShouldBe("This field resolver should never be called. It is only used to prevent the default field resolver from being used.");
```

Comment:
If now all the tests throw an exception, then what is the benefit of using `AutoRegisteringInterfaceGraphType`?

Reply:
All the resolver tests fail of course.  The class still builds all the fields and field arguments based on an interface, which is still very useful.  Attributes still work, except the ones that affect resolvers like `[Scoped]`.  Even `[FromServices]` is still useful as it will not generate an field argument for that method parameter (although `[Ignore]` is equally as useful for that scenario).

Reply:
Could one build `AutoRegisteringInterfaceGraphType` from interface and then modify resolvers to return something else?

Reply:
Well you can, but validation checks in the execution strategy will prevent interface resolvers from being used, whether it's based on `AutoRegisteringInterfaceGraphType` or `InterfaceGraphType`, since if it resolves to an object type (as we determined was required so that `__typename` works) then the object's resolver is used instead.

Reply:
One _may_ have a problem if the interface fields are returning now ( https://github.com/graphql-dotnet/graphql-dotnet/pull/3416#issuecomment-1324361913 ) from `SubFields` and for some reason he calls resolver manually.

Reply:
Migration notes added